contracts/peg/BasePSM.sol

```diff
@@ -0,0 +1,339 @@
+    function setReservesThreshold(uint256 newReservesThreshold)
+        external
+        override
+        onlyGovernorOrAdmin
+    {
+        _setReservesThreshold(newReservesThreshold);
+    }
+
+    /// @notice set the target for sending surplus reserves
+    function setSurplusTarget(IPCVDeposit newSurplusTarget)
+        external
+        override
+        onlyGovernorOrAdmin
+    {
+        _setSurplusTarget(newSurplusTarget);
+    }
+
+    /// @notice helper function to set reserves threshold
+    function _setReservesThreshold(uint256 newReservesThreshold) internal {
+        require(
+            newReservesThreshold > 0,
+            "PegStabilityModule: Invalid new reserves threshold"
+        );
+        uint256 oldReservesThreshold = reservesThreshold;
+        reservesThreshold = newReservesThreshold;
+
+        emit ReservesThresholdUpdate(
+            oldReservesThreshold,
+            newReservesThreshold
+        );
+    }
+
+    /// @notice helper function to set the surplus target
+    function _setSurplusTarget(IPCVDeposit newSurplusTarget) internal {
+        require(
+            address(newSurplusTarget) != address(0),
+            "PegStabilityModule: Invalid new surplus target"
+        );
+        require(
+            newSurplusTarget.balanceReportedIn() == address(underlyingToken),
+            "PegStabilityModule: Underlying token mismatch"
+        );
+
+        IPCVDeposit oldTarget = surplusTarget;
+        surplusTarget = newSurplusTarget;
+
+        emit SurplusTargetUpdate(oldTarget, newSurplusTarget);
+    }
+
+    // ----------- Public State Changing API -----------
+
+    /// @notice send any surplus reserves to the PCV allocation
+    function allocateSurplus() external override {
+        int256 currentSurplus = reservesSurplus();
+        require(
+            currentSurplus > 0,
+            "PegStabilityModule: No surplus to allocate"
+        );
+
+        _allocate(currentSurplus.toUint256());
+    }
+
+    /// @notice function to receive ERC20 tokens from external contracts
+    function deposit() external override {
+        int256 currentSurplus = reservesSurplus();
+        if (currentSurplus > 0) {
+            _allocate(currentSurplus.toUint256());
+        }
+    }
+
+    /// @notice function to redeem VOLT for an underlying asset
+    function redeem(
+        address to,
+        uint256 amountVoltIn,
+        uint256 minAmountOut
+    ) external virtual override whenNotPaused returns (uint256 amountOut) {
+        amountOut = getRedeemAmountOut(amountVoltIn);
+        require(
+            amountOut >= minAmountOut,
+            "PegStabilityModule: Redeem not enough out"
+        );
+
+        _beforeVoltRedeem(to, amountVoltIn, minAmountOut);
+
+        IERC20(volt()).safeTransferFrom(
+            msg.sender,
+            address(this),
+            amountVoltIn
+        );
+
+        SafeERC20.safeTransfer(underlyingToken, to, amountOut);
```

Comment:
Could you not do `underlyingToken.safeTransfer(to, amountOut)`?

Reply:
true, nice catch lool